### PR TITLE
[sdl2-ttf,tgui,sfml] Update sdl2-ttf to 2.20.0, revise dependencies import and export

### LINKED ISSUE #25661: [sdl2-ttf] update to 2.20.0
Library name:sdl2-ttf

New version number:2.20.0

Other information that may be useful (release notes, etc...)
https://github.com/libsdl-org/SDL_ttf/releases

### PR DESCRIPTION
**Pull request by dg0yt**
Reprise of  #25803 (@Thomas1664), #25713 (@FrankXie05). They made the major work.

- #### What does your PR fix?
  Fixes https://github.com/microsoft/vcpkg/issues/25661.
  Modernizes `tgui`, and updates import and export of dependencies.
  Fixes and updates import and export of dependencies in port `sfml`, in particular OpenGL, needed for building `tgui[tool]:x64-linux`.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  unchanged, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

### CODE REVIEW

**Comment by dg0yt** [comment]
- SFML is the default backend in TGUI. Using the SDL backend requires user to explicitly add 
   `#include <TGUI/Backends/SDL.hpp>`.
 - The TGUI pc file isn't installed for all platforms, the given lib name doesn't match the actual file name for most configurations, and dependencies are completely missing.

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/tgui/portfile.cmake [hunk 1]
```diff
@@ -1,20 +1,31 @@
+if(NOT "sdl2" IN_LIST FEATURES AND NOT "sfml" IN_LIST FEATURES)
```
[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/tgui/portfile.cmake:1]: Define "necessary", from user POV.

File: ports/tgui/usage [hunk 2]
```diff
@@ -0,0 +1,4 @@
+tgui provides CMake targets:
+
+    find_package(TGUI CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE tgui)
```
Comment (FrankXie05) [inline ports/tgui/usage:4]: It's not available.
image
````
CMake Error at E:/feature-test/vcpkg/scripts/buildsystems/vcpkg.cmake:824 (_find_package):
      Could not find a package configuration file provided by "TGUI" with any of
         the following names:    
            
           TGUIConfig.cmake
           tgui-config.cmake
````

File: ports/sdl2-ttf/usage [hunk 3]
```diff
@@ -0,0 +1,4 @@
+sdl2-ttf provides CMake targets:
+
+    find_package(SDL2_ttf CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE $<IF:$<TARGET_EXISTS:SDL2_ttf::SDL2_ttf>,SDL2_ttf::SDL2_ttf,SDL2_ttf::SDL2_ttf-static>)
```
[the remark this answers is not in the document]
Comment (FrankXie05) [reply ports/sdl2-ttf/usage:4]:
````
 CMake Error at E:/feature-test/vcpkg/scripts/buildsystems/vcpkg.cmake:824 (_find_package):
   Could not find a package configuration file provided by "SDL2_ttf" with any
   of the following names:
 
     SDL2_ttfConfig.cmake
     sdl2_ttf-config.cmake
 
   Add the installation prefix of "SDL2_ttf" to CMAKE_PREFIX_PATH or set
   "SDL2_ttf_DIR" to a directory containing one of the above files.  If
   "SDL2_ttf" provides a separate development package or SDK, be sure it has
   been installed.
````
image

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Comment by dg0yt** [comment]
@FrankXie05 Then please run with `CMAKE_FIND_DEBUG_MODE` enabled. I assume it is available, but not accepted due to some missing dependency.

File: ports/sdl2-ttf/usage (hunk 3 above)
[the remark this answers is not in the document]
Comment (FrankXie05) [reply ports/sdl2-ttf/usage:4]: Sorry, I got the matching system architecture wrong.

**Review by FrankXie05 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the updates! (And also @Thomas1664 and @FrankXie05 )

**Pull request merged**